### DISCUSSION TITLE
Sending success message in HTTP response only when TCP destination return ack back.

### DISCUSSION THREAD

**Discussion by BforAzure2** [nextgenhealthcare/connect#4813] [Q&A]
Hi Mirth Gurus, 

We have a scenario where we are receiving HTTP  in source (json) and sending it to a TCP/MLLP listener - Now the use case is, we want to send  success response back to HTTP only when destination (MLLP) send us ack, otherwise want to send error response back to HTTP.

Tried different ways, like tried to catch ack in destination transformer code and wrote a conditional response to send back to HTTP but could't achieve the purpose. Can we get the step by step process in Mirth to achieve this please?

**Comment by pacmano1** [comment]
Post your try (code). What you likely missed is the fact a response transformer will not fire unless it is set to datatype raw.

**Comment by pacmano1** [comment]
Also on the forums at: https://forums.mirthproject.io/forum/mirth-connect/support/180098-sending-success-message-in-http-response-only-when-tcp-destination-return-ack-back
